utils/priority_queue.py

```diff
@@ -0,0 +1,6 @@
+from queue import PriorityQueue as PQ
+
+
+class PriorityQueue(PQ):
+    def peek(self):
```

Comment:
Weird. I guess there's no peek function for Queues? Seems like there either should be or might be a reason why there isn't. Any thoughts?

Reply:
It doesn't exist in the basic `queue.PriorityQueue` that ships with python. This has been an acceptable workaround looks like for many years. because of the implementation, you are guaranteed the peek working for [0] but not for any further in the `queue` array.

Because the `PQ` API doesn't have peek, and because `get` pops it out, I think peeking is a requirement for our use case.